lib/order.rb

```diff
@@ -0,0 +1,51 @@
+require_relative 'menu'
+
+class Order
+
+  attr_reader :menu_display_class, :analyzed_order, :menu
+  ORDER_MESSAGE = 'Place order by entering the number of the dish, a comma, and the quantity, e.g. \'3, 4\'. New line for new item.  When order is complete simply press return again.'
+
+  def initialize(menu_display_class: MenuDisplay, menu: Menu.new)
+    @menu_display_class = menu_display_class
+    @menu = menu
+    @analyzed_order = Hash.new
+  end
+
+  def take_order
+    puts menu_display_class.new.list_menu
+    puts ORDER_MESSAGE
+    get_input
+  end
+
+  def get_input
+    loop do
```

Comment:
Could you shorten this function by using a different type of loop? Maybe try using a while loop.